esrally/client.py

```diff
@@ -134,23 +134,34 @@ def wait_for_rest_layer(es, max_attempts=20):
 
     :param es: Elasticsearch client to use for connecting.
     :param max_attempts: The maximum number of attempts to check whether the REST API is available.
-    :return: True iff Elasticsearch is available.
+    :return: True iff Elasticsearch's REST API is available.
     """
+    # assume that at least the hosts that we expect to contact should be available. Note that this is not 100%
+    # bullet-proof as a cluster could have e.g. dedicated masters which are not contained in our list of target hosts
+    # but this is still better than just checking for any random node's REST API being reachable.
```

Comment:
This is a very useful comment to have.

Question: would it be dangerous to trigger a sniff of the eligible http hosts (e.g. via a helper method in our EsClientFactory invoking [#elasticsearch.Transport#sniff_hosts()](https://elasticsearch-py.readthedocs.io/en/master/connection.html#elasticsearch.Transport.sniff_hosts))? I was thinking if we explicitly ask for a fresh list of hosts before the check, then no unavailable hosts should be reachable. Then the same call could be invoked before the load driver starts. The caveat with this approach would be that it could potentially override the explicitly list provided by `--target-hosts`. Thoughts?

Reply:
I'd try not to build any smartness into this? Without reading all of the involved code I don't think we can reason what nodes will be returned on cluster bootstrap (let's assume not all nodes are up yet or not all of them might have opened the HTTP port). I was even considering exposing an explicit command line parameter but thought that this would be a good compromise.

Reply:
Agreed. Let's keep things simple here.

For the record I had a look what the elasticsearch py client does when sniff gets invoked [here](https://github.com/elastic/elasticsearch-py/blob/a9cdb247001c40e94d916f58c14f87e4b281867c/elasticsearch/transport.py#L198-L238) and it collects a list of eligible http hosts via `/_nodes/_all/http`.